shared/models/gcp_artifact_provider.rb

```diff
@@ -0,0 +1,93 @@
+require_relative "artifact_provider"
+require_relative "artifact"
+require_relative "build"
+require_relative "project"
+
+require "google/cloud/storage"
+require "pathname"
+
+module FastlaneCI
+  # ArtifactProvider backed by a Google Cloud Platform Storage bucket.
+  class GCPStorageArtifactProvider < ArtifactProvider
+    class << self
+      # Provide a simple default root_path for users that don't want much configuration.
+      def root_browser
+        "https://console.cloud.google.com/storage/browser/"
+      end
+    end
+
+    # @return [String]
+    attr_accessor :cloud_project
+
+    # @return [String]
+    attr_accessor :json_keyfile_path
+
+    # @return [String]
+    attr_accessor :bucket_name
+
+    # @return [Google::Cloud::Storage]
+    attr_accessor :storage
+
+    # @return [Google::Cloud::Storage::Bucket]
+    attr_accessor :bucket
+
+    # @return [String] The class of the provider
+    attr_accessor :class_name
+
+    def initialize(cloud_project: nil, json_keyfile_path: nil, bucket_name: nil)
+      self.cloud_project = cloud_project
+      self.json_keyfile_path = json_keyfile_path
+      self.bucket_name = bucket_name
+      self.class_name = self.class.name.to_s
+    end
+
+    def init_storage!
+      # TODO: This will break when google-cloud-storage is updated.
```

Comment:
What do you mean this will break when ` google-cloud-storage is updated`? Why is that?

Reply:
So _fastlane_ depends on a certain version of `google-api-client` pretty outdated, and `google-cloud-storage` too, so I had to downgrade the version to match the requirements of _fastlane_, there is a PR open to update it, but it might last long before getting merged.

Reply:
Ahhh, let's add that comment into the code then, and you can specify exactly which versions you are talking about. That way if it does break we have some extra knowledge and context around it.